lib/identity_cache/query_api.rb

```diff
@@ -476,5 +486,16 @@ def was_new_record? # :nodoc:
       pk = self.class.primary_key
       !destroyed? && transaction_changed_attributes.has_key?(pk) && transaction_changed_attributes[pk].nil?
     end
+
+    def load_and_readonlyify(association_name)
+      record_or_records = send(association_name)
+
+      readonly_record_or_records = if self.class.reflect_on_association(association_name).collection?
+        record_or_records.map { |p| p.dup.tap(&:readonly!) }
+      else
+        record_or_records.dup.tap(&:readonly!) if record_or_records
+      end
+      readonly_record_or_records
```

Comment:
You don't need this variable.  You can just end the function on the if statement to return its result